custom_components/waste_collection_schedule/waste_collection_schedule/source/cannock_chase_dc_gov_uk.py

```diff
@@ -0,0 +1,89 @@
+import datetime
+import xml.etree.ElementTree as ET
+
+import requests
+from waste_collection_schedule import Collection
+from waste_collection_schedule.exceptions import SourceArgumentException
+
+TITLE = "Cannock Chase Council"
+DESCRIPTION = "Source for cannockchasedc.gov.uk services for Cannock Chase Council, UK."
+URL = "https://www.cannockchasedc.gov.uk"
+TEST_CASES = {
+    "Test_001": {"uprn": "100031640287", "postcode": "WS15 1DN"},
+    "Test_002": {"uprn": "100031640289", "postcode": "WS15 1DN"},
+    "Test_003": {"uprn": "100031624295", "postcode": "WS11 6DY"},
+    "Test_004": {"uprn": "10008163213", "postcode": "WS11 7UD"},
+}
+
+API_URL = "https://ccdc.opendata.onl/DynamicCall.dll"
+ICON_MAP = {
+    "REFUSE": "mdi:trash-can",
+    "RECYCLING": "mdi:recycle",
+    "GARDEN WASTE": "mdi:leaf",
+}
+
+HOW_TO_GET_ARGUMENTS_DESCRIPTION = {
+    "en": "You can find your UPRN by visiting https://www.findmyaddress.co.uk/ and entering in your address details.",
+}
+PARAM_DESCRIPTIONS = {
+    "en": {
+        "uprn": "An easy way to discover your Unique Property Reference Number (UPRN) is by going to https://www.findmyaddress.co.uk/ and entering in your address details.",
+        "postcode": "Postcode of the property",
+    },
+}
+
+SERVICE_NAME_MAP = {
+    "Refuse Collection Service": "Refuse",
+    "Recycle Collection Service": "Recycling",
+    "Garden Collection Service": "Garden waste",
+}
+
+
+class Source:
+    def __init__(self, uprn: str | int, postcode: str):
+        self._uprn: str = str(uprn).zfill(12)
+        self._postcode: str = postcode
+
+    def fetch(self) -> list[Collection]:
+        args = {
+            "Method": "CollectionDates",
+            "UPRN": self._uprn,
+            "Postcode": self._postcode,
+        }
+
+        r = requests.post(API_URL, data=args)
+        r.raise_for_status()
+
+        ns = {"ws": "http://webservices.whitespacews.com/"}
+        tree = ET.fromstring(r.text)
```

Comment:
I'm noting here that the `bandit` hook flagged:
> cannock_chase_dc_gov_uk.py:58: B314[bandit]: MEDIUM: Using xml.etree.ElementTree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace xml.etree.ElementTree.fromstring with its defusedxml equivalent function or make sure defusedxml.defuse_stdlib() is called

I was hesitant to add in a new dependency for `defusedxml` and can see a few existing uses of `xml.etree.ElementTree.fromstring`, but if you're ok with it, I can add it and replace all usages of `xml.etree.ElementTree.fromstring`?

Reply:
I don't think it's that big of a security concern here.